File: Cmdlets/AddCultureAndRegionInfoCommand.vb
Imports System.Globalization
Imports System.Management.Automation

<Cmdlet("Add", "CultureAndRegionInfo")>
Public Class AddCultureAndRegionInfoCommand
	Inherits CultureAndRegionInfoCommandBase

	<Parameter(Mandatory:=True)>
	Public Property Name() As String

	<Parameter(Mandatory:=True)>
	Public Property CultureAndRegionModifier() As CultureAndRegionModifiers


	Protected Overrides Sub BeginProcessing()
		Dim builder As New CultureAndRegionInfoBuilder(Name, CultureAndRegionModifier)

		CultureAndRegionInfoDictionary.Add(Name, builder)
	End Sub
End Class

File: WindowsCmdlets/Commands/AddCultureAndRegionInfoCommand.cs
using System.Globalization;
using System.Management.Automation;

namespace TIKSN.Cmdlets.Commands
{
    [Cmdlet("Add", "CultureAndRegionInfo")]
    public class AddCultureAndRegionInfoCommand : CultureAndRegionInfoCommandBase
    {
        [Parameter(Mandatory = true)]
        public CultureAndRegionModifiers CultureAndRegionModifier { get; set; }

        [Parameter(Mandatory = true)]
        public string Name { get; set; }

        protected override void BeginProcessing()
        {
            CultureAndRegionInfoBuilder builder = new CultureAndRegionInfoBuilder(Name, CultureAndRegionModifier);

            CultureAndRegionInfoDictionary.Add(Name, builder);
        }
    }
}